cpp/negentropy_wrapper.c

```diff
@@ -0,0 +1,229 @@
+#include <iostream>
+
+#include "negentropy.h"
+#include "negentropy/storage/BTreeMem.h"
+#include "negentropy_wrapper.h"
+
+//This is a C-wrapper for the C++ library that helps in integrating negentropy with nim code.
+//TODO: Do error handling by catching exceptions
+
+void printHexString(std::string_view toPrint){
+    for (size_t i = 0; i < toPrint.size(); ++i) {
+        printf("%0hhx", toPrint[i]);
+    }
+    printf("\n");
+}
+
+void* storage_new(const char* db_path, const char* name){
+    negentropy::storage::BTreeMem* storage;
+/* 
+    auto env = lmdb::env::create();
+    env.set_max_dbs(64);
+    env.open(db_path, 0);
+
+    lmdb::dbi btreeDbi;
+
+    {
+        auto txn = lmdb::txn::begin(env);
+        btreeDbi = negentropy::storage::BTreeMem::setupDB(txn, name);
+        txn.commit();
+    } */
+    //TODO: Finish constructor 
+    storage = new negentropy::storage::BTreeMem();
+    return storage;
+}
+
+void* negentropy_new(void* storage, uint64_t frameSizeLimit){
+    //TODO: Make these typecasts into macros??
+    negentropy::storage::BTreeMem* lmdbStorage;
+    //TODO: reinterpret cast is risky, need to use more safe type conversion.
+    lmdbStorage = reinterpret_cast<negentropy::storage::BTreeMem*>(storage);
+
+    Negentropy<negentropy::storage::BTreeMem>* ne;
+    try{
+    ne = new Negentropy<negentropy::storage::BTreeMem>(*lmdbStorage, frameSizeLimit);
+    }catch(negentropy::err e){
+        //TODO:Find a way to return this error
+        return NULL;
+    }
+    return ne;
+}
+
+size_t negentropy_initiate(void* negentropy, buffer* out){
+    Negentropy<negentropy::storage::BTreeMem>* ngn_inst;
+    ngn_inst = reinterpret_cast<Negentropy<negentropy::storage::BTreeMem>*>(negentropy);
+
+    std::string* output = new std::string();
```

Comment:
This should be deleted at the end with something like:
`delete output;`
In this particular case I think is better to keep `output` in the stack:
```suggestion
    std::string output;
```

Reply:
Good catch, missed this one. I can actually free this after the memcpy.